### PR TITLE
Added pybindings for DerivativeSpatialMetric in GR

### PR DESCRIPTION
**Pull request by ajosorio3**
## Proposed changes

Added pybinding for DerivativeSpatialMetric in GR.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/GeneralRelativity/Python/Bindings.cpp [hunk 1]
```diff
@@ -63,6 +64,13 @@ void bind_spacetime_impl(py::module& m) {  // NOLINT
 
 template <size_t Dim>
 void bind_impl(py::module& m) {  // NOLINT
+  m.def(
+      "derivative_inverse_spatial_metric",
```
Comment (nilsvu) [inline src/PointwiseFunctions/GeneralRelativity/Python/Bindings.cpp]: Please name this exactly like the C++ function: `deriv_inverse_spatial_metric` (it's easier to work with the pybindings if they mirror the C++ interface exactly)

**Review by nilsvu — CHANGES_REQUESTED** [review]

**Comment by nilsvu** [comment]
Did you intend to close this PR? You can just keep it open while you're making changes. The PR updates when you push to the branch.

**Comment by ajosorio3** [comment]
HI Nils, I noticed my proposed changes were for the develop branch instead of my own branch. I'll be making a new PR with your proposed changes and provide the link here

**Comment by nilsvu** [comment]
Ok!

**Comment by ajosorio3** [comment]
Here is the link for the new PR: https://github.com/sxs-collaboration/spectre/pull/5098

**Pull request closed without merging**